Tools/debug/CollectSystemInfo.ps1

```diff
@@ -0,0 +1,234 @@
+    }  catch {
+        Write-Warning "Failed to collect list of services: $_"
+    }
+}
+
+function Get-RunningProcesses {
+    try {
+        Get-Process | Select-Object -Property Id, ProcessName, StartTime | Export-Csv -Path RunningProcesses.csv -NoTypeInformation
+    } catch {
+        Write-Warning "Failed to collect list of running processes: $_"
+    }
+}
+
+function Get-InstalledApplications {
+    try {
+        Get-ItemProperty -Path 'HKLM:\Software\Microsoft\Windows\CurrentVersion\Uninstall\*' |
+        Select-Object -Property DisplayName, DisplayVersion, Publisher, InstallDate |
+        Export-Csv -Path InstalledApplications.csv -NoTypeInformation
+    } catch {
+        Write-Warning "Failed to collect list of installed applications: $_"
+    }
+}
+
+function Get-InstalledKBs {
+    try {
+        Get-HotFix | Select-Object -Property Description, HotFixID, InstalledOn | Export-Csv -Path InstalledKBs.csv -NoTypeInformation
+    } catch {
+        Write-Warning "Failed to collect list of installed KBs: $_"
+    }
+}
+
+function Get-WindowsMemoryDump {
+    $dumpFiles = @()
+    $memoryDumpPath = "$env:SystemRoot\MEMORY.DMP"
+    $minidumpPath = "$env:SystemRoot\Minidump"
+
+    try {
+        $dumpFiles += Get-Item -Path $memoryDumpPath -ErrorAction Stop
+    } catch {
+        Write-Host "No MEMORY.DMP file found or failed to access: $_"
+    }
+
+    try {
+        $minidumpFiles = Get-ChildItem -Path $minidumpPath -Filter "*.dmp" -ErrorAction Stop
+        $dumpFiles += $minidumpFiles.FullName
+    } catch {
+        Write-Host "No Minidump directory found or failed to access minidump files: $_"
+    }
+
+    if ($dumpFiles.Count -eq 0) {
+        Write-Host $dumpFiles
+    }
+
+    return $dumpFiles
+}
+
+function Remove-TempFiles {
+    param (
+        [string[]]$FilesToDelete
+    )
+    foreach ($file in $FilesToDelete) {
+        $resolvedFiles = Resolve-Path $file -ErrorAction SilentlyContinue
+        foreach ($resolvedFile in $resolvedFiles) {
+            try {
+                Remove-Item -Path $resolvedFile -Recurse -Force
+            } catch {
+                Write-Warning "Failed to delete file: $resolvedFile - $_"
+            }
+        }
+    }
+}
+
+function Write-InformationToArchive {
+    try {
+        $date = Get-Date -Format "yyyy-MM-dd_HH-mm-ss"
+        $archiveFile = "SystemInfo_$date.zip"
+        $filesToArchive = @('*.txt', '*.csv', '*.evtx', '*.log')
+
+        $collectedFiles = foreach ($pattern in $filesToArchive) {
+            Get-ChildItem -Path $pattern -ErrorAction SilentlyContinue
+        }
+
+        if (Test-Path -Path "LocaleMetaData") {
+            $collectedFiles += 'LocaleMetaData'
+        }        
+
+        if ($IncludeSensitiveData) { 
+            $additionalFiles = Get-WindowsMemoryDump
+            $collectedFiles += $additionalFiles
+        }
+        $collectedFiles = $collectedFiles | Where-Object { $_ -ne $null }
```

Comment:
Is `| Where-Object { $_ -ne $null }` really necessary?

Reply:
It's for preventing this error 
`WARNING: Failed to archive information: Cannot validate argument on parameter 'Path'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again`

Reply:
Why does it contain `$null`?